test/converters/xcresult3/converter_test.go

```diff
@@ -0,0 +1,74 @@
+package xcresult3
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-steplib/steps-deploy-to-bitrise-io/test/junit"
+	"github.com/stretchr/testify/require"
+)
+
+func copyTestdataToDir(t *testing.T, pathInTestdataDir, dirPathToCopyInto string) string {
+	err := command.CopyDir(
+		filepath.Join("../../testdata/", pathInTestdataDir),
+		dirPathToCopyInto,
+		false,
+	)
+	require.NoError(t, err)
+	return filepath.Join(dirPathToCopyInto, pathInTestdataDir)
+}
+
+func TestConverter_XML(t *testing.T) {
+	t.Run("xcresults3 success-failed-skipped-tests.xcresult", func(t *testing.T) {
```

Comment:
NOTE: what is the reason for running the single test case as a subtest?

Reply:
to be able to `defer` the temp directory cleanup

that said, as we only have a single test in it right now it doesn't really matter, but if we'd have another one this way we could generate and cleanup separate temp dirs for them